src/Security/Domain/Authentication/Model/AuthenticatedToken.php

```diff
@@ -0,0 +1,256 @@
+ *
+ */
+declare(strict_types=1);
+
+namespace Security\Domain\Authentication\Model;
+
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Role\Role;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * @package Security\Authentication
+ */
+class AuthenticatedToken implements TokenInterface
+{
+    /**
+     * @param string|\Stringable|UserInterface
+     */
+    private $user;
+
+    /**
+     * @param bool
+     */
+    private $isAuthenticated = false;
+
+    /**
+     * @var array
+     */
+    private $attributes = [];
+
+    /**
+     * @var Role[]
+     */
+    private $roles = [];
+
+    /**
+     * @var array
+     */
+    private $roleNames = [];
+
+    /**
+     * Token constructor.
+     *
+     * @param $roles Role[]|string[]
+     */
+    public function __construct($roles)
+    {
+        foreach ($roles as $role) {
+            if (\is_string($role)) {
+                $role = new Role($role, false);
+            } elseif (!$role instanceof Role) {
+                throw new \InvalidArgumentException(
+                    sprintf('$roles must be an array of strings, but got "%s".', \gettype($role))
+                );
+            }
+
+            $this->roles[] = $role;
+            $this->roleNames[] = (string) $role;
+        }
+    }
+
+    public function __serialize(): array
+    {
+        return [$this->user, $this->isAuthenticated, $this->roles, $this->attributes, $this->roleNames];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function serialize()
+    {
+        return $this->__serialize();
+    }
+
+    /**
+     * @param array<mixed> $data
+     */
+    public function __unserialize(array $data): void
+    {
+        [$this->user, $this->isAuthenticated, $this->roles, $this->attributes] = $data;
+
+        // migration path to 4.3+
+        if (null === $this->roleNames = $data[4] ?? null) {
+            $this->roleNames = [];
+            foreach ($this->roles as $role) {
+                $this->roleNames[] = (string) $role;
+            }
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function unserialize($serialized)
+    {
+        $this->__unserialize(\is_array($serialized) ? $serialized : unserialize($serialized));
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function __toString()
+    {
+        $class = static::class;
+        $class = substr($class, strrpos($class, '\\') + 1);
+
+        $roles = [];
+        foreach ($this->roles as $role) {
+            $roles[] = $role->getRole();
+        }
+
+        return sprintf(
+            '%s(user="%s", authenticated=%s, roles="%s")',
+            $class,
+            $this->getUsername(),
+            json_encode($this->isAuthenticated),
+            implode(', ', $roles)
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getRoles()
+    {
+        return [new Role('ROLE_ADMIN'), new Role('ROLE_LOLO')];
```

Comment:
Those roles are hardcoded and "Symfony\Component\Security\Core\Role\Role;" is Deprecated. 
Also this getRoles methods seems never called.

Should we remove it  or return an empty array to respect the interface contract?